### PR TITLE
Use deprecated property instead of stability level

### LINKED ISSUE #582: Disambiguate attribute deprecation
There are currently two ways to deprecate at attribute:
1. By setting stability to deprecated
    ```yaml
    - id: status_code
      brief: 'Deprecated, use `http.response.status_code` instead.'
      stability: deprecated
      ....
    ```
2. By providing `deprecated: description` property
    ```yaml
    - id: status_code
      deprecated: "use `http.response.status_code` instead."
    ```

The first approach is problematic: once attribute is deprecated it's no longer possible to know what was the stability at deprecation time. 
It's problematic for code generation:
- I want to separate experimental and stable attributes into different artifacts (or packages) based on stability level - I no longer know where to put deprecated ones (where they used to be)
- I want to drop deprecated attributes that were experimental at deprecation time and only keep ones that were stable at deprecation time - I'm not able to do it.

Related to #551

### PR DESCRIPTION
**Pull request by lmolkova**
Fixes #582

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* [x] ~~schema-next.yaml~~

### CODE REVIEW

**Review by arminru — APPROVED** [review]

**Review by pyohannes — APPROVED** [review]

File: model/registry/deprecated/http.yaml [hunk 1]
```diff
@@ -7,37 +7,37 @@ groups:
...
         type: string
         brief: 'Deprecated, use `url.path` and `url.query` instead.'
-        stability: deprecated
+        deprecated: "Split to `url.path` and `url.query."
         examples: ['/search?q=OpenTelemetry#SemConv']
       - id: request_content_length
         type: int
         brief: 'Deprecated, use `http.request.header.content-length` instead.'
-        stability: deprecated
+        deprecated: "Renamed to `http.request.header.content-length`."
```
Comment (trask) [inline model/registry/deprecated/http.yaml:35]: (I'm also good with current text both here and below calling it a rename)
```suggestion
        deprecated: "Replaced by `http.request.header.content-length`."
```

File: model/registry/deprecated/network.yaml [hunk 2]
```diff
@@ -7,47 +7,47 @@ groups:
...
       - id: peer.name
         type: string
-        stability: deprecated
+        deprecated: "Renamed to `server.address` on client spans and `client.address` on server spans."
         brief: Deprecated, use `server.address` on client spans and `client.address` on server spans.
         examples: ['example.com']
       - id: peer.port
         type: int
-        stability: deprecated
+        deprecated: "Renamed tp `server.port` on client spans and `client.port` on server spans."
```
Comment (trask) [inline model/registry/deprecated/network.yaml:30]:
```suggestion
        deprecated: "Renamed to `server.port` on client spans and `client.port` on server spans."
```

File: model/registry/deprecated/network.yaml [hunk 3]
```diff
@@ -7,47 +7,47 @@ groups:
...
         brief: Deprecated, use `server.port` on client spans and `client.port` on server spans.
         examples: [8080]
       - id: host.name
         type: string
-        stability: deprecated
+        deprecated: "Renamed to `server.address`."
```
Comment (trask) [inline model/registry/deprecated/network.yaml:35]:
```suggestion
        deprecated: "Replaced by `server.address`."
```

File: model/registry/deprecated/network.yaml [hunk 4]
```diff
@@ -7,47 +7,47 @@ groups:
...
         brief: Deprecated, use `server.address`.
         examples: ['example.com']
       - id: host.port
         type: int
-        stability: deprecated
+        deprecated: "Renamed to `server.port`."
```
Comment (trask) [inline model/registry/deprecated/network.yaml:40]:
```suggestion
        deprecated: "Replaced by `server.port`."
```

File: model/registry/deprecated/http.yaml [hunk 5]
```diff
@@ -7,37 +7,37 @@ groups:
...
         examples: 3495
       - id: response_content_length
         type: int
         brief: 'Deprecated, use `http.response.header.content-length` instead.'
-        stability: deprecated
+        deprecated: "Renamed to `http.response.header.content-length`."
```
Comment (trask) [inline model/registry/deprecated/http.yaml:40]:
```suggestion
        deprecated: "Replaced by `http.response.header.content-length`."
```

**Review by trask — APPROVED** [review]

File: model/registry/deprecated/http.yaml (hunk 1 above)
Comment (lmolkova) [reply model/registry/deprecated/http.yaml]: Thanks, I like `replaced by` better

**Comment by trask** [comment]
(I moved this to "Ready to be Merged", would probably be nice to get it into upcoming release)

**Review by reyang — APPROVED** [review]

**Comment by marcalff** [comment]
@lmolkova 

Please clarify if this causes a breaking change or not in the templates used to generate code.

In C++ for example, we use:

```
{%- if (attribute.stability | string()) == "StabilityLevel.DEPRECATED" %}
```

in here:

* https://github.com/open-telemetry/opentelemetry-cpp/blob/v1.13.0/buildscripts/semantic-convention/templates/SemanticAttributes.h.j2#L68-L70

What should that code fragment change to, once new semantic conventions are used ?

**Comment by lmolkova** [comment]
> @lmolkova
> 
> Please clarify if this causes a breaking change or not in the templates used to generate code.
> 
> In C++ for example, we use:
> 
> ```
> {%- if (attribute.stability | string()) == "StabilityLevel.DEPRECATED" %}
> ```
> 
> in here:
> 
> * https://github.com/open-telemetry/opentelemetry-cpp/blob/v1.13.0/buildscripts/semantic-convention/templates/SemanticAttributes.h.j2#L68-L70
> 
> What should that code fragment change to, once new semantic conventions are used ?

yes, you're right it does cause breaking change on the jinja template side :(
You'd have to change jinja template to do `{%- if attribute.deprecated %}` instead.

**Pull request merged**